scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.ApplePayEventReceiver.cs.erb

```diff
@@ -44,7 +48,22 @@ namespace <%= namespace %>.SDK.iOS {
         }
 
         public void DidFinishCheckoutSession(string serializedMessage) {
-            //TODO
+            var message = NativeMessage.CreateFromJSON(serializedMessage);
+            var paymentStatus = (NativePaymentStatus) Enum.Parse(typeof(NativePaymentStatus), (string) message.Content);
+            switch (paymentStatus) {
+            case NativePaymentStatus.Success:
+                OnSuccess();
+                return;
+            case NativePaymentStatus.Cancelled:
+                OnCancelled();
+                return;
+            case NativePaymentStatus.Failed:
+                var error = new ShopifyError(ShopifyError.ErrorType.CheckoutError, "Unable to retrieve a payment from the user's payment provider. Fallback to web checkout.");
+                OnFailure(error);
+                return;
+            default:
+                throw new NotImplementedException(string.Format("Unable to handle PaymentStatus: {0}", message.Content));
```

Comment:
I think an exception will actually be thrown by `Enum.Parse` and you'll never land in `default`.